### PR TITLE
Ethan branch pycharm

### PR DESCRIPTION
**Pull request by ethannieskes**
Sorry it took a while, I added a LOT of validation checks to the tests, with code to pass said tests.  I want to look more at views to see what we are doing there and adjust my methods/tests as needed

### CODE REVIEW

**Review by CAndreas7 — APPROVED** [review]
Looks good just needs a remerge to fix conflicts

File: Management/models.py [hunk 1]
```diff
@@ -40,6 +42,96 @@ class User(models.Model):
...
+            raise ValidationError("Name must be of type String")
+        if name.__len__() == 0:
+            raise ValueError("Name cannot be empty")
+
+        self.lName = name
+
+    def getfName(self):
+        return self.fName
+
+    def setfName(self, name):
```
Comment (SlinkyGoat) [inline Management/models.py:103]: No raising Error if name == None?

**Review by SlinkyGoat — APPROVED** [review]

File: Management/models.py [hunk 2]
```diff
@@ -40,6 +42,96 @@ class User(models.Model):
             (3, "TA")
         ))
 
+    def addAccount(self, email, lname, fname, password, phone, role):
+        if User.objects.filter(email=email).count() == 0:
+            if email is None or password is None or phone is None or role is None:
+                raise ValidationError("Email cannot be None")
```
[the remark this answers is not in the document]
Comment (ethannieskes) [reply Management/models.py:48]: I'll split that up in my next iteration, Ill have it in by our meeting

File: Management/models.py (hunk 1 above)
Comment (ethannieskes) [reply Management/models.py:103]: Problems noted, Ill fix by our meeting

**Pull request merged**